### ISSUE TITLE
UnusedPrivateMember should ignore `external` classes

### ISSUE THREAD

**Issue by ccjernigan** [detekt/detekt#4340]
The rule or changes you're looking for might have already been suggested!
Please search in the [issues](https://github.com/detekt/detekt/issues) before creating a new one.

## Expected Behavior of the rule
The `UnusedPrivateMember` rule should ignore `external` classes by default.

Example code:

```
@JsModule("@bugsnag/js")
@JsNonModule
external class Bugsnag {
    companion object {
        fun start(bugsnagParams: BugsnagParams)
        fun notify(error: Error)
    }
}

external interface BugsnagParams {
    val apiKey: String
    val appVersion: String
}
```

With Detekt 1.19, the actual behavior is that the Bugsnag companion object methods are flagged as unused.  The expected behavior is that this is not flagged.

## Context
In Kotlin/JS, `external` declarations are just a bridge to an existing Javascript API.  The Detekt rule is not actionable for these, and the only option is to suppress them.
